extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -164,112 +187,121 @@ private void restartJobs() {
         if (schedulerFuture != null && !schedulerFuture.isCancelled()) {
             schedulerFuture.cancel(true);
         }
-
-        schedulerFuture = scheduler.schedule(new Runnable() {
-            @Override
-            public void run() {
+        if (localExecutor != null) {
+            localExecutor.shutdownNow();
+        }
+        localExecutor = Executors.newSingleThreadExecutor();
```

Comment:
No, that would be even worse as we would spawn two additional threads, not just one.
I was under the impression that we are talking about a few milliseconds execution time for any of these jobs, so having them executed right here shouldn't really cause any delay.
But ok, to keep the "restartJobs" method clean, I can also live with having them submitted to be executed asynchronously. So let's leave it as it is.